src/plugins/fcrypt/fcrypt.c

```diff
@@ -213,14 +213,30 @@ static int encrypt (KeySet * pluginConfig, Key * parentKey)
 		if (rename (tmpFile, keyString (parentKey)) != 0)
 		{
 			ELEKTRA_SET_ERRORF (31, parentKey, "Renaming file %s to %s failed.", tmpFile, keyString (parentKey));
+			fprintf (stderr, "%s - rename %s -> %s failed\n", __FUNCTION__, tmpFile, keyString (parentKey));
 			result = -1;
 		}
+		else
+		{
+			fprintf (stderr, "%s - rename %s -> %s succeeded\n", __FUNCTION__, tmpFile, keyString (parentKey));
+		}
+	}
+
+	// always shred the temporary file
+	if (shredTemporaryFile (tmpFileFd, parentKey) != 1)
```

Comment:
Yes, seems like on success you shredder the wrong file ;)

You need to `open (keyString (parentKey))` before rename, and shredder the fd you get there. (the overwritten, old file)